File: scase-core/src/main/scala/io/jobial/scase/core/ReceiverClient.scala
package io.jobial.scase.core

import scala.concurrent.duration.FiniteDuration

trait ReceiverClient[F[_], M] {

  def receive: F[M]

  def receive(timeout: FiniteDuration): F[M]

  def receiveWithContext: F[MessageReceiveResult[F, M]]

  def receiveWithContext(timeout: FiniteDuration): F[MessageReceiveResult[F, M]]

  def stop: F[Unit]
}

File: scase-core/src/main/java/io/jobial/scase/core/javadsl/ReceiverClient.java
package io.jobial.scase.core.javadsl;

import cats.effect.IO;
import io.jobial.scase.core.RequestTimeout;

import java.util.concurrent.CompletableFuture;

import static io.jobial.scase.core.javadsl.JavaUtils.ioToCompletableFuture;

public class ReceiverClient<M> {

    private io.jobial.scase.core.ReceiverClient<IO, M> client;

    public ReceiverClient(io.jobial.scase.core.ReceiverClient<IO, M> client) {
        this.client = client;
    }

    public CompletableFuture<M> receive() throws RequestTimeout {
        return ioToCompletableFuture(client.receive());
    }

    public CompletableFuture<?> stop() {
        return ioToCompletableFuture(client.stop());
    }
}
